build.ps1

```diff
@@ -70,162 +75,163 @@ function Invoke-Configure {
-    $env:INCLUDE = 'C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE;C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE;C:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um;C:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\shared;C:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\um;C:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\winrt;'
-    $env:PATH = New-PathString -StartingPath $env:PATH -Path 'C:\Program Files (x86)\MSBuild\14.0\bin\amd64;C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64;C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\VCPackages;C:\WINDOWS\Microsoft.NET\Framework64\v4.0.30319;C:\WINDOWS\Microsoft.NET\Framework64\;C:\Program Files (x86)\Windows Kits\10\bin\x64;C:\Program Files (x86)\Windows Kits\10\bin\x86;C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6.1 Tools\x64\'
+function Invoke-Lint([string]$ToolChain) {
+    rustup component add clippy
+    $clippyArgs += Convert-ArrayToArgs -arg A -list (Get-Content $PSScriptRoot\test\unexamined_lints.txt)
+    $clippyArgs += Convert-ArrayToArgs -arg A -list (Get-Content $PSScriptRoot\test\allowed_lints.txt)
+    $clippyArgs += Convert-ArrayToArgs -arg W -list (Get-Content $PSScriptRoot\test\lints_to_fix.txt)
+    $clippyArgs += Convert-ArrayToArgs -arg D -list (Get-Content $PSScriptRoot\test\denied_lints.txt)
+
+    $clippyCommand = "$(Get-CargoCommand $ToolChain) clippy --all-targets --tests -- $clippyArgs"
+    Write-Host "--- Running clippy!"
+    Write-Host "Clippy rules: $clippyCommand"
+    Invoke-Expression $clippyCommand
 }
 
-$env:PATH                       = New-PathString -StartingPath $env:PATH    -Path 'C:\Program Files\7-Zip'
-$env:PATH                       = New-PathString -StartingPath $env:PATH    -Path $ChocolateyHabitatBinDir
-$env:LIB                        = New-PathString -StartingPath $env:LIB     -Path $ChocolateyHabitatLibDir
-$env:INCLUDE                    = New-PathString -StartingPath $env:INCLUDE -Path $ChocolateyHabitatIncludeDir
-$env:SODIUM_LIB_DIR             = $ChocolateyHabitatLibDir
-$env:LIBARCHIVE_INCLUDE_DIR     = $ChocolateyHabitatIncludeDir
-$env:LIBARCHIVE_LIB_DIR         = $ChocolateyHabitatLibDir
-$env:OPENSSL_LIBS               = 'ssleay32:libeay32'
-$env:OPENSSL_LIB_DIR            = $ChocolateyHabitatLibDir
-$env:OPENSSL_INCLUDE_DIR        = $ChocolateyHabitatIncludeDir
-$env:LIBZMQ_PREFIX              = Split-Path $ChocolateyHabitatLibDir -Parent
+function Get-CargoCommand($toolchain) {
+    "cargo +$toolchain"
+}
 
-Write-RustToolVersion
+If($Features) {
+    $featureString = "--features $Features"
```

Comment:
Is there special quoting we need to do for multiple features, either here or when invoking this script?

Is there a better way to accept multiple option values rather than (what I'm assuming is) a single string?